scripts/tasks/fluentui-publish.ts

```diff
@@ -1,33 +1,163 @@
-      throw new Error(result.error?.stack || `lerna publish failed with status ${result.status}`);
+    const branch = execCommandSync(gitRoot, 'git', ['branch', '--show-current']);
+    execCommandSync(gitRoot, 'git', ['fetch', 'origin']);
+    execCommandSync(gitRoot, 'git', ['reset', '--hard', `origin/${branch}`]); // sometimes lerna add gitHead in package.json after release
+
+    // sync fluent version
+    execCommandSync(gitRoot, 'yarn', ['syncpack:fix']);
+    const gitStatus = execCommandSync(gitRoot, 'git', ['status']);
+    if (!gitStatus.includes('nothing to commit, working tree clean')) {
+      execCommandSync(gitRoot, 'git', ['add', `\*package.json`]);
+      execCommandSync(gitRoot, 'git', ['commit', '-m', `"chore: fix dependencies after react-northstar release"`]);
+      execCommandSync(gitRoot, 'git', ['push']);
     }
+
+    // make sure there's no more than one fluent versions
+    execCommandSync(gitRoot, 'yarn', ['syncpack:list']);
+  };
+}
+
+// pack all public fluent ui packages, used by ci to store nightly built artifacts
+const PUBLIC_FLUENT_PACKAGES = [
+  'ability-attributes',
+  'accessibility',
+  'code-sandbox',
+  'docs-components',
+  'react-bindings',
+  'react-builder',
+  'react-component-event-listener',
+  'react-component-nesting-registry',
+  'react-component-ref',
+  'react-context-selector',
+  'react-icons-northstar',
+  'react-northstar-emotion-renderer',
+  'react-northstar-fela-renderer',
+  'react-northstar-styles-renderer',
+  'react-northstar',
+  'react-proptypes',
+  'react-telemetry',
+  'state',
+  'styles',
+];
+
+const TODAY = new Date().toISOString().split('T')[0]; // yyyy-mm-dd
+
+const replaceDepVersionWithNightlyUrl = (packageRoot, packageName) => {
+  const packageJson = require(`@fluentui/${packageName}/package.json`);
+  const dependencies = packageJson.dependencies || {};
+  let hasFluentDependency = false;
+  for (const key of Object.keys(dependencies)) {
+    if (key.startsWith('@fluentui/') && PUBLIC_FLUENT_PACKAGES.includes(fluentPackageName)) {
+      const fluentPackageName = key.substring('@fluentui/'.length);
+      hasFluentDependency = true;
+      dependencies[
+        key
+      ] = `https://fluentsite.blob.core.windows.net/nightly-builds/${TODAY}/fluentui-${fluentPackageName}-${packageJson.version}.tgz`;
+    }
+  }
+
+  if (hasFluentDependency) {
+    fs.writeFileSync(path.resolve(packageRoot, 'package.json'), JSON.stringify(packageJson, null, 2));
+  }
+};
+
+export function packFluentTarballs() {
+  return function() {
+    const gitRoot = findGitRoot();
+    const tempFolderForPacks = path.resolve(gitRoot, 'fluentui-nightly');
+    fs.mkdirSync(tempFolderForPacks);
+
+    const packPackage = (packageName: string) => {
+      const packageRoot = path.resolve(gitRoot, 'packages', 'fluentui', packageName);
+      replaceDepVersionWithNightlyUrl(packageRoot, packageName);
+      execCommandSync(tempFolderForPacks, 'npm', ['pack', packageRoot]);
```

Comment:
```suggestion
      execCommandSync(tempFolderForPacks, 'yarn', ['pack', packageRoot]);
```

nit, we still can use `yarn` 😏

Reply:
Somehow `yarn pack` just hangs. I couldn't figure out why. I saw there's github issue that `yarn pack` takes longer than `npm pack`. I'm going to keep it here